### DISCUSSION TITLE
Custom Language

### DISCUSSION THREAD

**Discussion by DavidNic11** [suren-atoyan/monaco-react#149] [Q&A] [answered]
I'm having some trouble getting the monaco instance (`monaco.init()`) to recognize the custom language example from [here](https://microsoft.github.io/monaco-editor/playground.html#extending-language-services-custom-languages). I was wondering if there was an example that showed something like it? If not, where should I initialize the monaco instance to get the editor to recognize the custom language?

**Comment by suren-atoyan** [comment, marked as answer]
Hello @DavidNic11 

By using `monaco` utility you can have access to monaco instance, which gives you an opportunity to define custom theme, language, etc.

[Here](https://codesandbox.io/s/monaco-editorreact---monaco-instance-5pbmx?fontsize=14&hidenavigation=1&theme=dark) is a snippet that demonstrates a usage of `monaco` utility
The equivalent of the [example](https://microsoft.github.io/monaco-editor/playground.html#extending-language-services-custom-languages) you've mentioned is [here](https://codesandbox.io/s/monaco-editorreactcustom-language-5yqj6?file=/src/index.js)

<hr />

I guess I should add a new section in README titled `custom language` 🙂

**Reply by DavidNic11** [reply]
@suren-atoyan Thanks for the quick response and the example! That worked like a charm. If you would like some help with documentation, I'd be more than happy to help

**Reply by suren-atoyan** [reply]
I'll make a draft and share it here. I'll be glad to get any feedback. Thank you 🙂
